### PR TITLE
Vk nv framebuffer mixed samples

### PR DESCRIPTION
**Pull request by cmayer1**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1664,6 +1693,70 @@ static bool ValidatePipelineUnlocked(layer_data *dev_data, std::vector<std::uniq
...
+                        "pCreateInfo[%d].pMultisampleState->rasterizationSamples (%u) is greater than the number of samples of the "
+                        "subpass color attachment (%u).",
+                        pipelineIndex, pipelineIndex, raster_samples, subpass_color_samples);
+                }
+
+                const auto *coverage_modulation_state = lvl_find_in_chain<VkPipelineCoverageModulationStateCreateInfoNV>(
+                    pPipeline->graphicsPipelineCI.pMultisampleState->pNext);
+
+                if (coverage_modulation_state && (coverage_modulation_state->coverageModulationTableEnable == VK_TRUE)) {
+                    if (coverage_modulation_state->coverageModulationTableCount != (raster_samples / subpass_color_samples)) {
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: This won't be protected from the `sub_color_samples == 0` if the `IsPowerOfTwo` test is removed...

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -1664,6 +1693,70 @@ static bool ValidatePipelineUnlocked(layer_data *dev_data, std::vector<std::uniq
...
+                    skip |= log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_PIPELINE_EXT,
+                                    HandleToUint64(pPipeline->pipeline), "VUID-VkGraphicsPipelineCreateInfo-subpass-01411",
+                                    "vkCreateGraphicsPipelines: pCreateInfo[%d].pMultisampleState->rasterizationSamples (%u) "
+                                    "does not match the number of samples of the RenderPass depth attachment (%u).",
+                                    pipelineIndex, raster_samples, subpass_depth_samples);
+                }
+            }
+        }
+
+        if (IsPowerOfTwo(subpass_color_samples)) {
```
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Not clear on the rational for the `IsPowerOfTwo` test here.  Clearly the parameter_validation layer should emit an error regarding for having more than one bit set in this (does it?), but it's not clear that one would want to silence these errors even in that case.

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -1638,6 +1638,35 @@ static bool ValidatePipelineUnlocked(layer_data *dev_data, std::vector<std::uniq
...
+        uint32_t subpass_num_samples = 0;
+
+        accumColorSamples(subpass_num_samples);
+
+        if (subpass_desc->pDepthStencilAttachment && subpass_desc->pDepthStencilAttachment->attachment != VK_ATTACHMENT_UNUSED) {
+            const auto attachment = subpass_desc->pDepthStencilAttachment->attachment;
+            subpass_num_samples |= static_cast<uint32_t>(pPipeline->rp_state->createInfo.pAttachments[attachment].samples);
+        }
+
+        if (subpass_num_samples && (!IsPowerOfTwo(subpass_num_samples) || (subpass_num_samples != raster_samples))) {
```
[the remark this answers is not in the document]
Comment (cmayer1) [reply layers/core_validation.cpp]: Sure, I can count in the prior loop how many attachments we have that are not VK_ATTACHMENT_UNUSED and then use attachmentCount != 0

File: layers/core_validation.cpp (hunk 1 above)
Comment (cmayer1) [reply layers/core_validation.cpp]: When removing the IsPowerofTwo, I can replace it by a if (color_attachment_count !=0). Would this be OK for you?

File: layers/core_validation.cpp (hunk 2 above)
Comment (cmayer1) [reply layers/core_validation.cpp]: When removing the IsPowerOfTwo test I end up having more than one possible value for the number of samples that I use to test against raster_samples. To perform the tests I would need to select one of these values. I can use the max as you suggested in the initial review or use a kind of worst case and choose either min or max depending on the VUID.
VUID 01412: if (raster_samples < subpass_max_color_samples) 
VUID 01415: if ((raster_samples > subpass_min_color_samples) && ..
Can you please let me know your preference ?

File: layers/core_validation.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: Okay, I understand better now why you did it that way.  Leave it as it is, and add a comment, okay?

File: layers/core_validation.cpp (hunk 2 above)
Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: If you can verify that the invalid case of having more than one possible value for the number of samples is caught elsewhere... then I'm fine with it as is.

File: layers/core_validation.cpp (hunk 1 above)
Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: Only if needed, see below.

File: layers/core_validation.cpp (hunk 2 above)
Comment (jzulauf-lunarg) [reply layers/core_validation.cpp]: Then okay as is.

File: layers/core_validation.cpp (hunk 3 above)
Comment (jzulauf-lunarg) [inline layers/core_validation.cpp]: Add a comment about how/when subpass_num_samples is zero (i.e. all attachements unused)

Comment (cmayer1) [reply layers/core_validation.cpp]: OK, I will add a comment and push a new (rebased) version.

**Pull request merged**